extensions/twenty/src/create-object-record.tsx

```diff
@@ -0,0 +1,117 @@
+import { Action, ActionPanel, Detail, Icon, List, showToast, Toast, useNavigation } from "@raycast/api";
+import { randomUUID } from "crypto";
+
+import twenty from "./services/TwentySDK";
+import { ObjectIcons } from "./enum/icons";
+import { OpenCreateObjectRecordForm } from "./pages";
+import { usePromise } from "@raycast/utils";
+import { useState } from "react";
+
+export default function CreateObjectRecord() {
+  const {
+    isLoading,
+    data: activeDataModels,
+    error,
+  } = usePromise(async () => {
+    const activeDataModels = await twenty.getActiveDataModels();
+
+    if (typeof activeDataModels === "string") {
+      throw new Error(activeDataModels as string);
+    }
+
+    return activeDataModels;
+  });
+
+  const [isOpenView, setIsOpenView] = useState(false);
+  const { push } = useNavigation();
+
+  if (error) {
+    return <Detail markdown={` # ERROR \n\n ${error.message}`} />;
+  }
+
+  const standardActiveModel = activeDataModels?.filter((model) => !model.isCustom);
+  const customActiveModel = activeDataModels?.filter((model) => model.isCustom);
+
+  return (
+    <List isLoading={isLoading} navigationTitle="Create Object Record" searchBarPlaceholder="Search Object Record">
+      <List.Section title="Standard Objects">
+        {standardActiveModel?.map((model) => {
+          const { id, description, labelPlural, icon } = model;
+          return (
+            <List.Item
+              id={id}
+              title={labelPlural}
+              subtitle={description ?? ""}
+              actions={
+                !isOpenView ? (
+                  <ActionPanel>
+                    <Action
+                      title="Create Record"
+                      icon={Icon.List}
+                      onAction={async () => {
+                        const objectRecordMetadata = await twenty.getRecordFieldsForDataModel(id);
+                        if (typeof objectRecordMetadata === "string") {
+                          await showToast({
+                            style: Toast.Style.Failure,
+                            title: objectRecordMetadata,
+                          });
+                        } else {
+                          setIsOpenView(true);
```

Comment:
A bit crude but it should get the job done - nice work 🥳 !
It's not working right now but we should be able to fix it by moving the `setOpenView(true)` above like so:
```suggestion
setIsOpenView(true);
                        const objectRecordMetadata = await twenty.getRecordFieldsForDataModel(id);
                        if (typeof objectRecordMetadata === "string") {
                          await showToast({
                            style: Toast.Style.Failure,
                            title: objectRecordMetadata,
                          });
                        } else {
                          
```

Reply:
Same for the 2nd action 👇 - we need to move the view otherwise it's not working.

Reply:
BEFORE:

https://github.com/user-attachments/assets/4d046dbc-272c-4234-ba35-0b3b2fbd9860

AFTER (moving the state to upper line - we now have `isLoading` and `Action` is hidden):

https://github.com/user-attachments/assets/08260cee-3dc6-464c-9ee8-fe95175ca906